src/Common/Bridge/BridgeMethodsInterfaceRule.php

```diff
@@ -0,0 +1,278 @@
+
+            return;
+        }
+
+        $firstInterface = $classNodeInterfaces[0];
+        $interfaceNode = new InterfaceNode($firstInterface);
+
+        $this->verifyInterface($node, $interfaceNode);
+    }
+
+    /**
+     * @param \PHPMD\Node\ClassNode $node
+     * @param \PHPMD\Node\InterfaceNode $interfaceNode
+     *
+     * @return void
+     */
+    protected function verifyInterface(ClassNode $node, InterfaceNode $interfaceNode): void
+    {
+        $bridgedInterfaceReflection = $this->getBridgedInterfaceReflection($node->getMethods());
+        if ($bridgedInterfaceReflection === null) {
+            $message = sprintf(
+                'The bridge is missing an interface. That violates the rule "%s"',
+                static::RULE,
+            );
+            $this->addViolation($interfaceNode, [$message]);
+
+            return;
+        }
+
+        $notMatchingMethods = $this->findNotMatchingMethodsForBridgeInterface($interfaceNode, $bridgedInterfaceReflection);
+
+        foreach ($notMatchingMethods as ['method' => $notMatchingMethod, 'error' => $errorMsg]) {
+            $message = sprintf(
+                'The bridge interface has incorrect method signature for \'%s()\'. %s That violates the rule "%s"',
+                $notMatchingMethod->getName(),
+                $errorMsg,
+                static::RULE,
+            );
+            $this->addViolation($interfaceNode, [$message]);
+        }
+    }
+
+    /**
+     * @param \PHPMD\Node\InterfaceNode $interfaceNode
+     * @param \ReflectionClass $bridgedInterfaceReflection
+     *
+     * @return array|null
+     */
+    protected function findNotMatchingMethodsForBridgeInterface(InterfaceNode $interfaceNode, ReflectionClass $bridgedInterfaceReflection): ?array
+    {
+        $notMatchingMethods = [];
+
+        foreach ($interfaceNode->getMethods() as $interfaceMethod) {
+            if (!$bridgedInterfaceReflection->hasMethod($interfaceMethod->getName())) {
+                $notMatchingMethods[] = [
+                    'method' => $interfaceMethod,
+                    'error' => 'Method does not exist in parent.',
+                ];
+
+                continue;
+            }
+
+            $bridgedInterfaceReflectionMethod = $bridgedInterfaceReflection->getMethod($interfaceMethod->getName());
+
+            $interfaceMethodName = sprintf('%s::%s', $interfaceNode->getFullQualifiedName(), $interfaceMethod->getName());
+            $interfaceMethodReflection = new ReflectionMethod($interfaceMethodName);
+
+            $errors = $this->compareBridgeAndParentMethodInterface($interfaceMethodReflection, $bridgedInterfaceReflectionMethod);
+
+            $interfaceMethodReturnType = $this->reflactionReturnTypeToString($interfaceMethodReflection);
+            $parentInterfaceReturnType = $this->getReturnTypeFromParentInterface($interfaceMethod, $bridgedInterfaceReflection);
+
+            if (!$interfaceMethodReturnType) {
```

Comment:
so we are forcing to add return types in bridge interfaces?

Reply:
We are at least making it visible now if they are missing

Reply:
Yes.